tests/testcustomlinecap.c

```diff
@@ -0,0 +1,571 @@
+#ifdef WIN32
+#ifndef __cplusplus
+#error Please compile with a C++ compiler.
+#endif
+#endif
+
+#if defined(_WIN32)
+#include <Windows.h>
+#include <GdiPlus.h>
+
+#pragma comment(lib, "gdiplus")
+#else
+#include <GdiPlusFlat.h>
+#endif
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "testhelpers.h"
+
+#ifdef WIN32
+using namespace Gdiplus;
+using namespace DllExports;
+#endif
+
+static void verifyCustomLineCap (GpCustomLineCap *cap, LineCap expectedBaseCap, REAL expectedBaseInset)
+{
+	assert (cap && "Expected cap to be initialized.");
+
+	GpStatus status;
+	LineCap baseCap;
+	REAL baseInset;
+	LineCap startCap;
+	LineCap endCap;
+	LineJoin strokeJoin;
+	REAL widthScale;
+	CustomLineCapType type;
+
+	// GDI+ returns 0 here each time. This looks like a GDI+ bug.
+#if !defined(_WIN32)
```

Comment:
should probably be `!defined(USE_WINDOWS_LIBGDIPLUS)` right? I see a few more of these, we can do a separate cleanup pass.